suite-native/graph/src/graphMiddleware.ts

```diff
@@ -0,0 +1,15 @@
+import { createMiddleware } from '@suite-common/redux-utils';
+import { deviceActions } from '@suite-common/wallet-core';
+
+import { removeAccountGraphTimeframe } from './slice';
+
+export const graphMiddleware = createMiddleware((action, { next, dispatch }) => {
+    if (deviceActions.forgetDevice.match(action)) {
+        const deviceState = action.payload.state;
+        if (deviceState) {
+            dispatch(removeAccountGraphTimeframe({ deviceState }));
+        }
+    }
```

Comment:
Does this have to be middleware? You could just use this action in the reducer and change state

Reply:
great point! 6827addbe681dc197ce7761978dfd5b8249cf20b